sphinx-jsonschema/__init__.py

```diff
@@ -48,11 +155,21 @@ class JsonSchema(Directive):
                    'auto_reference': flag,
                    'auto_target': flag,
                    'timeout': float,
-                   'encoding': directives.encoding}
+                   'encoding': directives.encoding,
+                   'hide_key': hide_key,
```

Comment:
What is the format for hide_key?
Why do you need the csv module?

Reply:
Quoted strings separated with a comma, that's also the reason for the csv module to handle the parsing.

Strings themselves should be mostly compatible with JSON pointer, except they also support the `*` and `**` wildcards where:
`*` - Shallow search through one level
`**` - Deep search through multiple levels

In my project I'm using:

```
    :hide_key: "/**/examples","/**/_comment"
```

To remove all `examples` and `_comment` fields anywhere in the JSON document.
I'm also using:

```
    :hide_key_if_empty: "/**/default"
```

To hide `default` fields anywhere in the document if the default value is empty.

I could, for example do:
```
    :hide_key: "/$defs/**/_comment"
```
To remove `_comment` fields from anything that descends from `$defs`. 
Or:
```
    :hide_key: "/$defs/*/description"
```

To hide the `description` field of anything directly defined in `$defs`.

Reply:
I see. Do you really need the quotes?
Without those I suppose that split() and strip() would be sufficient.
I'm just trying to understand the necessity of csv.

Reply:
Ah right, good question, I'm hardwired to quoting everything. The quotes are not necessary unless you want to have the separator in the string itself, consider the following:

```
    :hide_key: /$defs/air, earth, fire, and water/
```

`air, earth, fire, and water` is a valid JSON key, but if we did split() with `,` separator we'd parse it as 4 different paths. The csv parser takes care of that - it can deal with both unquoted and quoted input.

Reply:
understood, thanks for the explanation.